File: mod_10_paradigmas_e_padroes_de_projeto/exemplos/oo/Java/Cilindro/src/entidades/Cilindro.java
package entidades;

public class Cilindro {
    private double raio;
    private double altura;

    public Cilindro(double diametro, double altura) {
        if (diametro <= 0 || altura <= 0) {
            throw new IllegalArgumentException("Diâmetro e altura devem ser valores positivos.");
        }
        this.raio = diametro / 2;
        this.altura = altura;
    }

    public double calcularVolume() {
        return Math.PI * Math.pow(raio, 2) * altura;
    }

    public double calcularAreaSuperficial() {
        return 2 * Math.PI * raio * (raio + altura);
    }

    public void mostrarDados() {
        System.out.printf("Diâmetro: %.2f cm, Altura: %.2f cm%n", raio * 2, altura);
    }
}

File: mod_10_paradigmas_e_padroes_de_projeto/exemplos/oo/Python/Cilindro/entidades/cilindro.py
import math

class Cilindro:
    def __init__(self, diametro: float, altura: float):
        if diametro <= 0 or altura <= 0:
            raise ValueError("Diâmetro e altura devem ser valores positivos.")
        self.raio = diametro / 2
        self.altura = altura

    def calcular_volume(self) -> float:
        return math.pi * (self.raio ** 2) * self.altura  # Volume em cm³

    def calcular_area_superficial(self) -> float:
        return 2 * math.pi * self.raio * (self.raio + self.altura)

    def mostrar_dados(self):
        print(f"Diâmetro: {self.raio * 2:.2f} cm, Altura: {self.altura:.2f} cm")
